pkg/api/v1/status/atlascluster.go

```diff
@@ -72,7 +99,28 @@ func AtlasClusterMongoDBVersionOption(mongoDBVersion string) AtlasClusterStatusO
 
 func AtlasClusterConnectionStringsOption(connectionStrings *mongodbatlas.ConnectionStrings) AtlasClusterStatusOption {
 	return func(s *AtlasClusterStatus) {
-		cs := ConnectionStrings(*connectionStrings)
+		pe := make([]PrivateEndpoint, 0, len(connectionStrings.PrivateEndpoint))
+		for _, v := range connectionStrings.PrivateEndpoint {
+			endpoints := make([]Endpoint, 0, len(v.Endpoints))
+			for _, e := range v.Endpoints {
+				endpoints = append(endpoints, Endpoint(e))
+			}
+
+			pe = append(pe, PrivateEndpoint{
```

Comment:
[q] why cant' we build the `PrivateEndpoint` directly from the Atlas object?

Reply:
Because it contains `[]mongodbatlas.Endpoint`, while our struct contains `[]status.Endpoint`. Even though the two `Endpoints` are directly convertible, their parent structs are not.

Reply:
what if the atlas-client and we add some field but forget to add the instruction here? 
Maybe we can serialize the `mongodbatlas.ConnectionStrings` into bytes and then deserialize into our object?

Reply:
You're probably right - I'll use JSON the same way we do in controller